src/apirequest.ts

```diff
@@ -317,7 +318,14 @@ async function createAPIRequestAsync<T>(parameters: APIRequestParams) {
   // now void.  This may be a source of confusion for users upgrading from
   // version 24.0 -> 25.0 or up.
   if (authClient && typeof authClient === 'object') {
-    return (authClient as OAuth2Client).request<T>(options);
+    if (options.http2) {
+      const authHeaders = await authClient.getRequestHeaders(options.url);
```

Comment:
This makes me wonder if this is the right place to do it. Wouldn't it be more logical to move all `http2` request logic to `gaxios`, and make `OAuth2Client.request` aware of `http2` option?

Reply:
I have absolutely had the same thought.  The other problem with the way this is implemented right now is that doesn't have retry support that's native to gaxios.  This started small, and by the time I finished I landed where you did.

@bcoe what do you think?  Only concern is complicating things too early before we get a few eyes on the codebase.

Reply:
I'm less convinced that gaxios is the right place for this code. It feels like we'd then be in a similar position to where we landed with `teeny-request`, where every feature that works for `http` is expected to work for `http2` in gaxios.

So, treating http2 as a drop in replacement for gaxios makes more sense to me...

I think that adding it into the library itself right now is reasonable, with an eye for eventually breaking it out into its own library potentially.

Reply:
Sounds good.  I like putting it here as a start, and then seeing where things go.